### PR TITLE
Add check for dependencies to ensure required dependencies are installed.

### PR DESCRIPTION
**Pull request by vishnoianil**
Signed-off-by: Anil Kumar Vishnoi <avishnoi@redhat.com>

### CODE REVIEW

File: hack/kind/kind.sh [hunk 1]
```diff
@@ -1,18 +1,48 @@
...
+    if [ -n "$1" ]; then
+        echo -e "[\033[0;31m FAILED \033[0m] ] ${1}"
+    fi
+}
 
 up() {
+    if ! [ -x "$(command -v kubectl)" ]; then
+        error_message "kubectl is required. Please install kubectl"
+        exit 1
+    fi
```
Comment (russellb) [inline hack/kind/kind.sh:34]: Might as well check for `kind` while you're at it

Comment (vishnoianil) [reply hack/kind/kind.sh:34]: I added that initially, but removed it, because kind itself does that. It throws `kind not found`.

Comment (vishnoianil) [reply hack/kind/kind.sh:34]: My above comment is hilarious 🗡️ , `Kind` throws `'kind not found` :). I meant it's the first command for up and down both and shell throws kind not found if it doesn't exist, so i thought it's redundant. But no harm in  explicit check.

Comment (vishnoianil) [reply hack/kind/kind.sh:39]: @russellb added explicit check for kind.

**Pull request merged**